### PR TITLE
EpicID error handling

### PR DESCRIPTION
**Pull request by belmirp**
Returning `ErrorResponse `instead of only message when `InvalidEpicIDException `is thrown:

Examples:

`{
  "errorCode": "errors.com.epicgames.account.id_not_32_characters",
  "errorMessage": "ID needs to be 32 characters long.",
  "messageVars": [
    "9c808754479eaa36e7efe896f204da1"
  ],
  "numericErrorCode": 18009,
  "originatingService": null,
  "intent": null,
  "error_description": null,
  "error": null
}`

`{
  "errorCode": "errors.com.epicgames.account.id_not_hexadecimal",
  "errorMessage": "ID needs to be a hexadecimal string.",
  "messageVars": [
    "9c808754479eaa36e7efe896f204da1_"
  ],
  "numericErrorCode": 18008,
  "originatingService": null,
  "intent": null,
  "error_description": null,
  "error": null
}`

Related: [#22](https://github.com/timiimit/UT4MasterServer/issues/22)

### CODE REVIEW

File: UT4MasterServer/Other/EpicID.cs [hunk 1]
```diff
@@ -27,14 +27,14 @@ private EpicID(string id)
 	{
 		if (id.Length != 32)
 		{
-			throw new InvalidEpicIDException("ID needs to be 32 characters long.");
+			throw new InvalidEpicIDException("ID needs to be 32 characters long.", id, 18009, "errors.com.epicgames.account.id_not_32_Characters");
```
Comment (Saibamen) [inline UT4MasterServer/Other/EpicID.cs:30]:
```suggestion
			throw new InvalidEpicIDException("ID needs to be 32 characters long.", id, 18009, "errors.com.epicgames.account.id_not_32_characters");
```

Comment (belmirp) [reply UT4MasterServer/Other/EpicID.cs]: Copy-paste error, fixed it.

File: UT4MasterServer/Controllers/ErrorsController.cs [hunk 2]
```diff
@@ -20,10 +21,14 @@ public IActionResult Index()
 		{
```
Comment (Saibamen) [inline UT4MasterServer/Controllers/ErrorsController.cs:21]: Can you also add logging exception here? `logger.LogError(ex)`

Comment (belmirp) [reply UT4MasterServer/Controllers/ErrorsController.cs:30]: Added, here is an example:
image

File: UT4MasterServer/Controllers/ErrorsController.cs [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
 	[HttpGet]
 	public IActionResult Index()
 	{
 		var message = "Internal server error occurred.";
 		var statusCode = 500;
 
 		var exceptionHandlerFeature = HttpContext.Features.Get<IExceptionHandlerPathFeature>();
+		var exception = exceptionHandlerFeature?.Error;
+
+		logger.LogError(exception, "Internal server error occurred.");
```
Comment (Saibamen) [inline UT4MasterServer/Controllers/ErrorsController.cs:27]: We already have this string in `message` variable.
```suggestion
		logger.LogError(exception, message);
```

Comment (belmirp) [reply UT4MasterServer/Controllers/ErrorsController.cs:27]: I know, but VS is complaining about that:
image

I will store the text in a constant.

File: UT4MasterServer/Exceptions/InvalidEpicIDException.cs [hunk 4]
```diff
@@ -5,7 +5,47 @@ namespace UT4MasterServer.Exceptions;
...
-	public InvalidEpicIDException(string message) : base(message) { }
+	private string _id { get; set; }
+	public string ID { get { return _id; } }
 
-	protected InvalidEpicIDException(SerializationInfo serializationInfo, StreamingContext streamingContext) : base(serializationInfo, streamingContext) { }
+	private int _numericErrorCode { get; set; }
+	public int NumericErrorCode { get { return _numericErrorCode; } }
+
+	private string _errorCode { get; set; }
+	public string ErrorCode { get { return _errorCode; } }
```
Comment (timiimit) [inline UT4MasterServer/Exceptions/InvalidEpicIDException.cs:15]: Is there a reason for having a set of private/public properties? public property could just have setter marked as private

File: UT4MasterServer/Other/EpicID.cs [hunk 5]
```diff
@@ -27,14 +27,14 @@ private EpicID(string id)
 	{
 		if (id.Length != 32)
 		{
-			throw new InvalidEpicIDException("ID needs to be 32 characters long.");
+			throw new InvalidEpicIDException("ID needs to be 32 characters long.", id, 18009, "errors.com.epicgames.account.id_not_32_characters");
```
Comment (timiimit) [inline UT4MasterServer/Other/EpicID.cs:30]: are the numerical error codes taken from somewhere or did you come up with them?
i think it should be the same as original when `errors.com.epicgames.modules.stats.invalid_account_id` or similar error is triggered on official servers

**Review by timiimit — APPROVED** [review]

**Pull request merged**